sonar-helm-for-iac/converters/tree_converter.go

```diff
@@ -0,0 +1,283 @@
+		nodeAsMessage = convertContinueNode(*node.(*parse.ContinueNode))
+	case *parse.DotNode:
+		nodeAsMessage = convertDotNode(*node.(*parse.DotNode))
+	case *parse.FieldNode:
+		nodeAsMessage = convertFieldNode(*node.(*parse.FieldNode))
+	case *parse.IdentifierNode:
+		nodeAsMessage = convertIdentifierNode(*node.(*parse.IdentifierNode))
+	case *parse.IfNode:
+		nodeAsMessage = convertIfNode(*node.(*parse.IfNode))
+	case *parse.ListNode:
+		nodeAsMessage = convertListNode(*node.(*parse.ListNode))
+	case *parse.NilNode:
+		nodeAsMessage = convertNilNode(*node.(*parse.NilNode))
+	case *parse.NumberNode:
+		nodeAsMessage = convertNumberNode(*node.(*parse.NumberNode))
+	case *parse.PipeNode:
+		nodeAsMessage = convertPipeNode(*node.(*parse.PipeNode))
+	case *parse.RangeNode:
+		nodeAsMessage = convertRangeNode(*node.(*parse.RangeNode))
+	case *parse.StringNode:
+		nodeAsMessage = convertStringNode(*node.(*parse.StringNode))
+	case *parse.TemplateNode:
+		nodeAsMessage = convertTemplateNode(*node.(*parse.TemplateNode))
+	case *parse.TextNode:
+		nodeAsMessage = convertTextNode(*node.(*parse.TextNode))
+	case *parse.VariableNode:
+		nodeAsMessage = convertVariableNode(*node.(*parse.VariableNode))
+	case *parse.WithNode:
+		nodeAsMessage = convertWithNode(*node.(*parse.WithNode))
+	default:
+		nodeAsMessage = &org_sonarsource_iac_helm.Node{
+			// NodeType duplicates declarations in parse.node, but with Unknown as 0
+			NodeType: (org_sonarsource_iac_helm.NodeType)(int(node.Type()) + 1),
+			Pos:      int64(node.Position()),
+		}
+	}
+	return nodeAsMessage
+}
+
+func convertActionNode(node parse.ActionNode) proto.Message {
+	return &org_sonarsource_iac_helm.ActionNode{
+		NodeType: org_sonarsource_iac_helm.NodeType_NodeAction,
+		Pos:      int64(node.Pos),
+		Pipe:     convertPipeNode(*node.Pipe),
+	}
+}
+
+func convertBoolNode(node parse.BoolNode) proto.Message {
+	return &org_sonarsource_iac_helm.BoolNode{
+		NodeType: org_sonarsource_iac_helm.NodeType_NodeBool,
+		Pos:      int64(node.Pos),
+		True:     node.True,
+	}
+}
+
+func convertBranchNode(node parse.BranchNode) proto.Message {
+	var elseList *org_sonarsource_iac_helm.ListNode
+	if node.ElseList == nil {
+		elseList = nil
+	} else {
+		elseList = convertListNode(*node.ElseList)
+	}
+	return &org_sonarsource_iac_helm.BranchNode{
+		Pos:      int64(node.Pos),
+		Pipe:     convertPipeNode(*node.Pipe),
+		List:     convertListNode(*node.List),
+		ElseList: elseList,
+	}
+}
+
+func convertBreakNode(node parse.BreakNode) proto.Message {
+	return &org_sonarsource_iac_helm.BreakNode{
+		NodeType: org_sonarsource_iac_helm.NodeType_NodeBreak,
+		Pos:      int64(node.Pos),
+	}
+}
+
+func convertChainNode(node parse.ChainNode) proto.Message {
+	return &org_sonarsource_iac_helm.ChainNode{
+		NodeType: org_sonarsource_iac_helm.NodeType_NodeChain,
+		Pos:      int64(node.Pos),
+		Field:    node.Field,
+	}
+}
+
+func convertCommandNode(node parse.CommandNode) *org_sonarsource_iac_helm.CommandNode {
```

Comment:
I'd say that it should always be `proto.Message`, because there is no need to know the specific type after conversion, and we can directly feed these data into protobuf engine. However, sometimes we actually need to know the specific type, for example when converting a list of variables in `pipe` node, which are then assigned to a field that expects `VariableNode`s